src/App.jsx

```diff
@@ -33,20 +96,60 @@ function App() {
       content: newTask
     };
 
-    setColumns({
-      ...columns,
-      todo: {
-        ...columns.todo,
-        tasks: [...columns.todo.tasks, task]
-      }
-    });
+    const firstColumnId = Object.values(columns)
+      .sort((a, b) => a.order - b.order)[0]?.id;
+
+    if (firstColumnId) {
+      setColumns({
+        ...columns,
+        [firstColumnId]: {
+          ...columns[firstColumnId],
+          tasks: [...columns[firstColumnId].tasks, task]
+        }
+      });
+    }
     setNewTask('');
   };
 
+  const handleDeleteColumn = (columnId) => {
+    const newColumns = { ...columns };
+    delete newColumns[columnId];
+    setColumns(newColumns);
+  };
+
   const onDragEnd = (result) => {
-    const { source, destination } = result;
+    const { source, destination, type } = result;
+    
     if (!destination) return;
 
+    // Se for arrasto de colunas
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
<button 
  className="delete-column"
  onClick={() => handleDeleteColumn(column.id)}
  aria-label={`Delete ${column.title} column`}
>
  ×
</button>
```
Consider using a more semantic HTML element for the delete button and add aria-label for accessibility.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;